libs/librrgraph/src/base/check_rr_graph.cpp

```diff
@@ -494,13 +497,13 @@ void check_rr_node(int inode, enum e_route_type route_type, const DeviceContext&
             //Don't worry about disconnect PINs which have no adjacent channels (i.e. on the device perimeter)
             bool check_for_out_edges = true;
             if (rr_type == IPIN || rr_type == OPIN) {
-                if (!has_adjacent_channel(rr_graph.rr_nodes()[inode], device_ctx.grid)) {
+                if (!has_adjacent_channel(rr_graph, grid, rr_graph.rr_nodes()[inode])) {
```

Comment:
Just a mark, rr_nodes() just expose the internal node storage directly to client function, which should be avoid. I suggest to rework this function later.

https://github.com/verilog-to-routing/vtr-verilog-to-routing/blob/7989197469e9650e45a00e035f67139edc296831/libs/librrgraph/src/base/rr_graph_view.h#L442-L445